File: Examples/test-suite/perl5/li_cmalloc_runme.pl
use strict;
use warnings;
use Test::More tests => 3;
BEGIN { use_ok('li_cmalloc') }
require_ok('li_cmalloc');

my $p = li_cmalloc::malloc_int();
li_cmalloc::free_int($p);

eval {
  my $p = li_cmalloc::calloc_int(-1);
  li_cmalloc::free_int($p);
};
isnt($@, undef);

File: Examples/test-suite/javascript/li_cmalloc_runme.js
var li_cmalloc = require("li_cmalloc");

var p = li_cmalloc.malloc_int(1);
li_cmalloc.free_int(p);

var ok = false;
try {
    p = li_cmalloc.calloc_int(-1);
    li_cmalloc.free_int(p);
} catch {
    ok = true;
}

if (!ok) {
    throw new Error;
}
